ilc/server/tailor/fetch-template.spec.js

```diff
@@ -0,0 +1,64 @@
+const chai = require('chai');
+const sinon = require('sinon');
+
+const fetchTemplates = require('./fetch-template');
+
+describe('fetch templates', () => {
+  const configsInjector = {
+    inject: sinon.stub(),
+  };
+
+  const newrelic = {
+    setTransactionName: sinon.spy(),
+  };
+
+  const registryService = {
+    getTemplate: (arg) => arg,
+  };
+
+  const request = {
+    router: {},
+    ilcState: 'ilcState text'
+  };
+
+  const parseTemplate = sinon.stub();
+
+  request.router.getFragmentsTpl = function(arg) {
+    return `${arg}`
+  };
+
+  it('should throw Error if template is undefined', async () => {
+    request.router.getRoute = function() {
+      return {
+        template: undefined,
+      }
+    };
+
+    await chai.expect(fetchTemplates({}, {}, registryService)(request)).to.eventually.rejectedWith('Can\'t match route base template to config map');
+  });
+
+  it('should set transaction name in newrelic', async () => {
```

Comment:
Pls test also case with special route and properly test logic of transaction naming (to make sure that regex works correctly in various cases)